scrapy/utils/project.py

```diff
@@ -39,16 +41,22 @@ def project_data_dir(project='default'):
         os.makedirs(d)
     return d
 
+
 def data_path(path, createdir=False):
-    """If path is relative, return the given path inside the project data dir,
-    otherwise return the path unmodified
+    """
+    Return the given path joined with the .scrapy data directory.
+    If given an absolute path, return it unmodified.
     """
     if not isabs(path):
-        path = join(project_data_dir(), path)
+        if inside_project():
+            path = join(project_data_dir(), path)
+        else:
+            path = join('.scrapy', path)
```

Comment:
so, before it raised NotConfigured and now it creates a directory in `CWD/.scrapy/<path>`.

And the goal is to allow httpcache and deltafetch (and similars) to run even outside a scrapy project.

It is a bit backwards incompatible, scripts running in readonly filesystem now will fail because directory can't be created.